internal/controller/datadogagent/global/fips.go

```diff
@@ -21,111 +21,124 @@ import (
 	"github.com/DataDog/datadog-operator/internal/controller/datadogagent/object/configmap"
 	"github.com/DataDog/datadog-operator/internal/controller/datadogagent/object/volume"
 	"github.com/DataDog/datadog-operator/pkg/controller/utils/comparison"
+	"github.com/DataDog/datadog-operator/pkg/defaulting"
 	"github.com/DataDog/datadog-operator/pkg/kubernetes"
 )
 
 // applyFIPSConfig applies FIPS related configs to a pod template spec
-func applyFIPSConfig(logger logr.Logger, manager feature.PodTemplateManagers, dda *v2alpha1.DatadogAgent,
-	resourcesManager feature.ResourceManagers) {
+func applyFIPSConfig(logger logr.Logger, manager feature.PodTemplateManagers, dda *v2alpha1.DatadogAgent, resourcesManager feature.ResourceManagers) {
 	globalConfig := dda.Spec.Global
 	fipsConfig := globalConfig.FIPS
 
-	// Add FIPS env vars to all containers except System Probe
-	for _, cont := range manager.PodTemplateSpec().Spec.Containers {
-		if cont.Name != string(apicommon.SystemProbeContainerName) {
-			manager.EnvVar().AddEnvVarToContainer(apicommon.AgentContainerName(cont.Name), &corev1.EnvVar{
-				Name:  DDFIPSEnabled,
-				Value: "true",
-			})
-			manager.EnvVar().AddEnvVarToContainer(apicommon.AgentContainerName(cont.Name), &corev1.EnvVar{
-				Name:  DDFIPSPortRangeStart,
-				Value: strconv.Itoa(int(*fipsConfig.Port)),
-			})
-			manager.EnvVar().AddEnvVarToContainer(apicommon.AgentContainerName(cont.Name), &corev1.EnvVar{
-				Name:  DDFIPSUseHTTPS,
-				Value: apiutils.BoolToString(fipsConfig.UseHTTPS),
-			})
-			manager.EnvVar().AddEnvVarToContainer(apicommon.AgentContainerName(cont.Name), &corev1.EnvVar{
-				Name:  DDFIPSLocalAddress,
-				Value: *fipsConfig.LocalAddress,
-			})
+	if apiutils.BoolValue(fipsConfig.ModeEnabled) {
+		// Add -fips suffix to each container image
+		for i, container := range manager.PodTemplateSpec().Spec.Containers {
+			// Note: if an image override is configured, this image tag will be overwritten
+			manager.PodTemplateSpec().Spec.Containers[i].Image = container.Image + defaulting.FIPSTagSuffix
 		}
+		for i, container := range manager.PodTemplateSpec().Spec.InitContainers {
+			manager.PodTemplateSpec().Spec.InitContainers[i].Image = container.Image + defaulting.FIPSTagSuffix
+		}
+		return
 	}
 
-	// Configure FIPS container
-	fipsContainer := getFIPSProxyContainer(fipsConfig)
-
-	image := common.GetImage(fipsConfig.Image, globalConfig.Registry)
-	fipsContainer.Image = image
-	if fipsConfig.Image.PullPolicy != nil {
-		fipsContainer.ImagePullPolicy = *fipsConfig.Image.PullPolicy
-	}
-
-	// Add FIPS container to pod
-	found := false
-	for i, cont := range manager.PodTemplateSpec().Spec.Containers {
-		if cont.Name == fipsContainer.Name {
-			manager.PodTemplateSpec().Spec.Containers[i] = fipsContainer
-			found = true
+	if apiutils.BoolValue(fipsConfig.ProxyEnabled) || apiutils.BoolValue(fipsConfig.Enabled) {
```

Comment:
it is difficult to tell, but the only change in this section is this new `if` that's wrapped around the existing code